embeddings/task/lightning_task/lightning_task.py

```diff
@@ -59,20 +63,28 @@ def best_validation_score(self) -> Optional[float]:
     def fit(
         self,
         data: HuggingFaceDataModule,
+        run_name: Optional[str] = None,
     ) -> None:
         if not self.model:
             raise self.MODEL_UNDEFINED_EXCEPTION
 
-        callbacks: List[Callback] = []
+        callbacks: List[Callback] = [
+            ModelCheckpoint(dirpath=self.output_path.joinpath("checkpoints"))
+        ]
         if "validation" in data.load_dataset().keys():
             callbacks.append(BestEpochCallback())
             callbacks.append(EarlyStopping(**self.early_stopping_kwargs))
 
         self.trainer = pl.Trainer(
-            default_root_dir=str(self.output_path), callbacks=callbacks, **self.task_train_kwargs
+            default_root_dir=str(self.output_path),
+            callbacks=callbacks,
+            logger=self.get_lightning_loggers(run_name, **self.logging_kwargs),
+            **self.task_train_kwargs
         )
         try:
             self.trainer.fit(self.model, data)
+            if "test" in data.load_dataset().keys():
+                self.trainer.test(self.model, data.test_dataloader())
```

Comment:
`test` run shouldn't be called from the `fit` method; we have a separate method `predict` for that with custom evaluator

Reply:
I do not agree. It performs a test epoch (and logs metrics of the test subset). It require access to self.trainer that is deleted at the end of .fit().

Reply:
The `fit()` method only fits the model to the data, **testing the model is undesired action in this context**; if we wanted to include this method in our code it should be implemented in another method such as fit_predict - but since we manually specify the predict method I do not see the reason of including the test() method from Trainer